gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java

```diff
@@ -31,127 +33,90 @@
 
-        project.getExtensions()
-                .getByType(JavaPluginExtension.class)
+        javaPluginExtension
                 .getSourceSets()
-                .getByName(SourceSet.MAIN_SOURCE_SET_NAME)
-                .resources(resources -> {
-                    SourceDirectorySet sourceDir = project.getObjects()
-                            .sourceDirectorySet("metricSchema", "metric schema")
-                            .srcDir(metricSchemaDir);
-                    sourceDir.include("metric-schema/**");
-                    resources.source(sourceDir);
+                .named(SourceSet.MAIN_SOURCE_SET_NAME)
+                .configure(sourceSet -> {
+                    sourceSet.getJava().srcDir(generateMetricsTask);
+                    sourceSet.getResources().srcDir(compileMetricSchemaTask.map(CompileMetricSchemaTask::getOutputDir));
                 });
 
-        return compileMetricSchemaTask;
-    }
-
-    private static SourceDirectorySet createSourceSet(Project project) {
-        SourceDirectorySet sourceSet =
-                project.getObjects().sourceDirectorySet("metricSchema", "Metric Schema source set");
-        sourceSet.srcDir("src/main/metrics");
-        sourceSet.include("**/*.yml");
-        return sourceSet;
-    }
+        project.getPluginManager().withPlugin("com.palantir.sls-java-service-distribution", _plugin -> {
+            project.getPluginManager().apply(MetricSchemaMarkdownPlugin.class);
+        });
 
-    private static void configureJavaSource(
-            Project project, TaskProvider<GenerateMetricSchemaTask> generateMetricSchemaTask) {
-        project.getExtensions()
-                .getByType(JavaPluginExtension.class)
-                .getSourceSets()
-                .getByName(SourceSet.MAIN_SOURCE_SET_NAME)
-                .getJava()
-                .srcDir(generateMetricSchemaTask);
-    }
+        configureIdea(project, generatedJavaDir, generatedResourcesDir);
 
-    private static void configureEclipse(Project project, TaskProvider<? extends Task> generateMetricsTask) {
-        project.getPluginManager().withPlugin("eclipse", _plugin -> {
-            try {
-                project.getTasks().named("eclipseClasspath").configure(t -> t.dependsOn(generateMetricsTask));
-            } catch (UnknownDomainObjectException e) {
-                // eclipseClasspath is not always registered
-            }
-        });
+        configureProjectDependencies(project);
     }
 
     private static void configureIdea(
-            Project project, TaskProvider<? extends Task> generateMetricsTask, Provider<Directory> outputDir) {
+            Project project, Provider<Directory> generatedJavaDir, Provider<Directory> generatedResourcesDir) {
         project.getPluginManager().withPlugin("idea", _plugin -> {
-            project.getTasks().named("ideaModule").configure(task -> task.dependsOn(generateMetricsTask));
```

Comment:
We can just remove the IDE task dependencies. We only support building projects using Gradle (which both Eclipse and IntelliJ support) and Gradle has all the proper dependencies.